src/ui/CategoryFacet/CategoryValue.ts

```diff
@@ -0,0 +1,99 @@
+import { Dom, $$ } from '../../utils/Dom';
+import { CategoryFacetTemplates } from './CategoryFacetTemplates';
+import { CategoryJsonValues } from './CategoryValue';
+import { CategoryChildrenValueRenderer } from './CategoryValueChildrenRenderer';
+
+export interface CategoryValueParent {
+  categoryChildrenValueRenderer: CategoryChildrenValueRenderer;
+  getPath: (partialPath?: string[]) => string[];
+}
+
+export type CategoryJsonValues = { [key: string]: string[] };
+
+export class CategoryValue implements CategoryValueParent {
+  private element: Dom;
+  private collapseArrow: Dom;
+  private arrowOnClick: (e: Event) => void;
+  private captionOnClick: (e: Event) => void;
+
+  public categoryChildrenValueRenderer: CategoryChildrenValueRenderer;
+
+  constructor(
+    private parentElement: Dom,
+    private value: string,
+    private parent: CategoryValueParent,
+    private categoryFacetTemplates: CategoryFacetTemplates
+  ) {
+    this.element = this.categoryFacetTemplates.buildListElement(this.value);
+    this.collapseArrow = this.categoryFacetTemplates.buildCollapseArrow();
+    this.categoryChildrenValueRenderer = new CategoryChildrenValueRenderer(this.element, categoryFacetTemplates, this);
+
+    this.arrowOnClick = () => {
+      this.closeChildMenu();
+    };
+    this.captionOnClick = () => {
+      this.openChildMenu();
+    };
+    this.getCaption().on('click', this.captionOnClick);
```

Comment:
fat arrow, or we can lose the proper `this` context.

Or `this.captionOnClick.bind(this)`

Reply:
The `captionOnClick` attribute already refers to a functions wrapped by a fat arrow function. I quickly tested and I think it looks like it should be sufficient. The goal of directly using the attribute as the event handler for the click event was to be able to have a direct reference to the event handler. 

This lets me use the `off` function from the `Dom` utilities and remove the event handler correctly from any reference which avoids a memory leak.